src/elements/Comments/NewComment.jsx

```diff
@@ -0,0 +1,55 @@
+import { useEffect, useState } from "react";
+import { postCommentById } from "../../utils/api";
+import { useParams } from "react-router-dom";
+
+export const NewComment = ({ username, setComments }) => {
+  const { id } = useParams();
+
+  const [postingComment, setPostingComment] = useState(false);
+  const [newCommentText, setNewCommentText] = useState("");
+
+  useEffect(() => {
+    setPostingComment(false);
+    setNewCommentText("");
+  }, []);
+
+  const submitComment = (event) => {
+    event.preventDefault();
+
+    if (newCommentText) {
+      setPostingComment(true);
+
+      postCommentById(id, newCommentText, username)
+        .then(({ comment: resComment }) => {
+          setComments((currentComments) => [...currentComments, resComment]);
+
+          setNewCommentText("");
+          setPostingComment(false);
+        })
+        .catch((error) => {
+          setPostingComment(false);
+        });
+    }
+  };
+
+  return (
```

Comment:
Apart from just disabling the button, a message would also be nice to let the user know that the comment is being posted.